### ISSUE TITLE
Escape key doesn't close OverflowMenu

### ISSUE THREAD

**Issue by lauraalkhoury** [IBM/carbon-components-react#1733]
## Detailed description

> Describe in detail the issue you're having. 

Escape key doesn't close the OverflowMenu component

> Is this issue related to a specific component?

Yes, OverflowMenu

> What did you expect to happen? What happened instead? What would you like to see changed?

Expected overflow menu to close if it's open and I press the escape key
Instead, it didn't close
Change request: make it close on esc keypress

> What browser are you working in?

Chrome Version 71.0.3578.98 (Official Build) (64-bit)

> What version of the Carbon Design System are you using?

versions:
“react”: “^16.4.0",
“carbon-components”: “^9.12.0",
“carbon-components-react”: “^6.28.2",
“carbon-icons”: “^7.0.7"

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

Continuous Delivery Pipeline Service - we need to have all accessibility defects fixed by Jan 31 

## Steps to reproduce the issue

1. Go to http://react.carbondesignsystem.com/?selectedKind=OverflowMenu
2. Open overflow menu
3. Press esc

**Comment by dakahn** [comment]
Hello! Thanks so much for pointing this out. I'll assign this to myself and get a fix pushed up asap.

**Comment by carmacleod** [comment]
Thanks, @dakahn !
Please note that you will want to call `event.stopPropagation();` and possibly also `event.preventDefault();` when handling the escape key so that it does not cause problems such as the one reported in https://github.com/IBM/carbon-components/issues/1398

**Comment by lauraalkhoury** [comment]
@dakahn also worth noting that when the OverflowMenu is open, putting focus back on the icon that opens/closes and hitting `space` doesn't close it either. Should i open a separate issue for that?

**Comment by carmacleod** [comment]
@lauraalkhoury The user shouldn't be able to use the keyboard to put focus back on the trigger, so no need for a separate issue, because the problem will "go away" after https://github.com/IBM/carbon-components-react/issues/1742 is fixed.

**Comment by lauraalkhoury** [comment]
Ah, makes sense! Thanks @carmacleod

**Comment by carmacleod** [comment]
Nice! Thanks!

**Comment by lauraalkhoury** [comment]
Thanks!! Will update today to pick up the fix
